sys/zuse.hoon

```diff
@@ -7942,4 +7936,148 @@
       --
     --
   --
+:: |dawn: pre-boot request/response de/serialization and validation
+::
+++  dawn
+  =>  |%
+      :: +live: public network state of a ship
+      ::
+      +$  live  (unit [=life breach=?])
+      :: +seed: private boot parameters
+      ::
+      +$  seed  [who=ship lyf=life key=ring sig=(unit oath:pki:jael)]
+      --
+  |%
+  :: |give:dawn: produce requests for pre-boot validation
+  ::
+  ++  give
+    =/  tract  ships:contracts:constitution:ethe
+    |%
+    :: +czar:give:dawn: Eth RPC for galaxy table
+    ::
+    ++  czar
+      ^-  octs
+      %-  as-octt:mimes:html
+      %-  en-json:html
+      %-  batch-read-request:ethereum
+      %+  turn  (gulf 0 255)
+      |=(gal=@ [`(scot %ud gal) tract ['getKeys(uint32)' [%uint gal]~]])
+    :: +hull:give:dawn: Eth RPC for ship's contract state
+    ::
+    ++  hull
+      |=  who=ship
+      ^-  octs
+      %-  as-octt:mimes:html
+      %-  en-json:html
+      (read-request:ethereum [~ tract ['ships(uint32)' [%uint `@`who]~]])
+    :: +turf:give:dawn: Eth RPC for network domains
+    ++  turf
+      ^-  octs
+      %-  as-octt:mimes:html
+      %-  en-json:html
+      %-  batch-read-request:ethereum
+      %+  turn  (gulf 0 2)
+      |=(idx=@ [`(scot %ud idx) tract ['dnsDomains(uint256)' [%uint idx]~]])
+    --
+  :: |take:dawn: parse responses for pre-boot validation
+  ::
+  ++  take
+    =,  dejs:format
```

Comment:
Probably also `=, constitution:ethereum`.

Reply:
As above, I expect this to need some refactoring.